### DISCUSSION TITLE
[已解決] 輸入注音問題

### DISCUSSION THREAD

**Discussion by msn0981** [vChewing/vChewing-macOS#252] [Q&A] [answered]
請問如果要出入注音，例如：我想打「ㄋ一ˇ ㄏㄠˇ」有辦法一次打完在按ENTER嗎？
一口氣打完ㄋ一ˇ或ㄏㄠˇ選字欄都沒有注音可以選，請問有什麼方法可以輸入注音嗎？

謝謝

**Comment by ShikiSuen** [comment, marked as answer]
@msn0981 抱歉，我沒太搞清楚你的需求。
你要打注音文的話，可以一次性敲「你好」這兩個字，然後在 Enter 遞交之前使用 Ctrl+Command+Enter，就可以改為遞交注音文了。

**Reply by msn0981** [reply]
感謝解答！
那另外想提問，如果是要打無法拼成字的注音，如ㄋㄧㄚˋ這類的，因為打完ㄋㄧㄚ再按ˋ會直接跳掉，是否只能一個一個打？

**Reply by ShikiSuen** [reply]
@msn0981 對。還真得這樣。目前我暫時沒想到好的辦法（因為會牽扯到對使用者體驗的綜合調整，得煩惱「怎樣設計才好」）。

**Reply by msn0981** [reply]
了解，感謝解答！

**Comment by ShikiSuen** [comment]
@msn0981 威注音現在可以「直接」輸入無法拼成字的注音了，方法參見近期的版本更新日誌。

**Reply by msn0981** [reply]
感謝通知！馬上來試看看，謝謝！

**Reply by msn0981** [reply]
剛試了一下，可以了！感謝新增此功能👍

**Reply by CollieIsCute** [reply]
@ShikiSuen  請問有相關連結可以查閱嗎？找了一下好像沒找到＠＠
目前是使用許氏鍵盤，也有想輸入注音文的需求，因此找到這則討論。
我知道 ctrl+command+enter 的輸法，但許氏注音有兩個注音重疊在同一個按鍵的情況，因此想知道有沒有什麼選擇注音的方法，謝謝。

**Reply by ShikiSuen** [reply]
@CollieIsCute 那抱歉了，許氏鍵盤的話真的沒辦法支援。
同理，任何無法與物理鍵盤完全一一映射的佈局，都不支援。比如酷音大千26鍵、倚天26鍵、星光，等。

https://twitter.com/vChewingIME/status/1651827974994534400

**Reply by CollieIsCute** [reply]
好，謝謝。我目前是輸入特定的字再刪掉一部分達成一樣的功能。比如我要打「ㄏ」，我就打「ㄏㄜ」再刪掉「ㄜ」
但此方法就目前無法在例如 github 網站上（會直接送出這段 comment ）或者某些其他電腦應用程式（如 telegram desktop 會無法遞交）。

**Reply by ShikiSuen** [reply]
@CollieIsCute 有些設計就是魚與熊掌不可兼得的。
你可以跟小麥那邊聊聊，看他們是否認為這種功能在原理上能做出來。
不過威注音這邊注定會用另外的方法就是了，因為兩邊各自用的注拼引擎完全不同。

GitHub 的問題呢，你可以事先開一個 TextEdit 視窗，將要貼的文字事先寫好，再藉由剪貼簿中轉。我就一直這麼做。
瀏覽器對熱鍵的響應行為，不是輸入法能夠預測的，所以這種與軟體行為判定有關的衝突不可避免。

**Reply by CollieIsCute** [reply]
@ShikiSuen 目前就自動選字、使用介面等等，我比較喜歡威注音。
維護成本太高的話，小問題可接受

**Reply by ShikiSuen** [reply]
@CollieIsCute 我還有一個設想，是一套組合拳：

一、在啟用動態注音鍵盤佈局的時候，聲介韻調四個位置各自多一個功能：記錄此時的十進位 charCode。

二、輸入法在注拼引擎內存有尚未完成拼寫的注音的時候「敲了 Ctrl+Command+Enter」的話，會叫出一個特殊的函式、將注拼引擎當中的 charCode 拽出來，這個 charCode 陣列會被拓印成專用的注音檢索陣列（檢索用關鍵詞是「`_COMPOUND_PARSERNAME_CHARCODE`」）。

三、組字引擎按需載入一個「專門用來處理複合注音排列的」語言模組，負責將上文的「charCode 檢索用關鍵詞」翻譯成「注音符號陣列」。比如說許氏鍵盤 C 對應「ㄕㄒ」、且 C 的小寫字母的十進位 CharCode 是 99，那麼檢索用關鍵詞「_COMPOUND_HSU_99」就會返回陣列「ㄕㄒ」。

四、（繼續上文「二」部分）將這個「注音檢索用關鍵詞陣列」插入組字引擎，然後讓組字引擎爬軌（walk），就可以從上文「三」的語言模組當中取得資料。此時，注音符號就像是被敲入組字區一樣，但你**可以叫出選字窗來選擇這個鍵位對應的注音符號**。

五、上文那樣敲「Ctrl+Command+Enter」的時候，如果組字區原本就是空的、且此時的注音排列為「注音與鍵盤按鍵一對一對應」的排列的話，就會直接遞交這份尚未完成拼寫的讀音。

以上應該是一套可行的設計原理，但是實裝起來要花費比較多的功夫。我這十年來都只用大千傳統排列，本身沒什麼動力去做這個功能。如果我今年下半年或者明年過年有時間的話，可能會實作一下看看。

**Reply by CollieIsCute** [reply]
@ShikiSuen 聽起來是個超讚的功能～好期待！

**Reply by ShikiSuen** [reply]
@CollieIsCute 但這樣就會將組字引擎搞得空前複雜，所以很頭痛XD。

**Reply by ShikiSuen** [reply]
@CollieIsCute 另外，波浪鍵符號選單也可以輸入注音，就是會麻煩一些。
